exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/partitionsender/PartitionSenderRootExec.java

```diff
@@ -231,7 +231,7 @@ public boolean innerNext() {
   }
 
   @VisibleForTesting
-  protected void createPartitioner() throws SchemaChangeException {
+  protected synchronized void createPartitioner() throws SchemaChangeException {
```

Comment:
Why you need to synchronized`createParitioner`? It's being called from `innerNext` which is already synchronized.

Reply:
Removed